### DISCUSSION TITLE
Question: Remote control of iOS

### DISCUSSION THREAD

**Discussion by vin-cf** [faaxm/spix#49] [Q&A]
Hi there, this is such a useful library by the way 🏆 . I'm getting a lot of mileage out of it on desktop platforms. 

Controlling the UI of an iOS app is something I am interested in too. I notice that the README mentions "Remote control, also of embedded devices / iOS". I've looked in the `examples/` folder but can't find any iOS-specific hints. Would it be possible to get a more detailed pointers in the README on how to get set up on iOS? In particular, I am wondering if:  

- anything specific needs to be done in the compilation of the iOS app with including the Spix library  
- What does a feasible set up for sending RPC to iOS look like? Is an iPad connected via USB a requirement? Or is it possible do this with the app running in [XCode's iOS simulator](https://developer.apple.com/documentation/xcode/running-your-app-in-the-simulator-or-on-a-device)?

I notice you mentioned in another discussion that you "use mostly linux and macOS", if you don't have access to iOS hardware I am happy to try this out myself with some pointers, reporting back with what works.

**Comment by faaxm** [comment]
It would indeed be great to have more documentation and tutorials on particular use cases. I am mostly maintaining spix in my private time, so it's usually not easy to get to improving the documentation. However, if you find a nice workflow for you, I would really appreciate you writing it down and contributing it to the repo. Maybe this could be the beginning of the much needed `docs/` directory ;-)

Of course to use spix with iOS, you have to have a Qt-based iOS app as the native UIKit is not supported. There you can integrate spix just as you would for a desktop app.
When running the rpc server in that app, it will open a tcp port and the test script could connect remotely through WiFi, as long as you know the ip address of your phone.
In the example test script
https://github.com/faaxm/spix/blob/d260b51a4702a753455d1d677f4aabad15e1d281/examples/RemoteCtrl/script/testScript.py#L3
you would replace `localhost` with the phone's ip address.
This should also work when running the ios app in the simulator as long as it allows you have an open tcp server running. Note that you probably don't want to expose your phone to the public internet while running the test server, as it doesn't have any password protection and would allow anyone to send commands to your app.
